sentence.go

```diff
@@ -66,6 +67,11 @@ func (s BaseSentence) String() string { return s.Raw }
 // parseSentence parses a raw message into it's fields
 func parseSentence(raw string) (BaseSentence, error) {
 	raw = strings.TrimSpace(raw)
+	tagBlock, raw, err := parseTagBlock(raw)
```

Comment:
The `BaseSentence.Raw` field should still contain the entire line that was passed into `parseSentence`. `parseTagBlock`  doesn't need to return the `raw` parameter.

Reply:
@icholy it appears the purpose of passing `raw` through `parseTagBlock()` is to strip the tag block so that the rest of `parseSentence()` doesn't choke on it. The code errors out here if `raw` doesn't contain an un-tagged sentence: https://github.com/xerra-eo/go-nmea/blob/4b3b9d4ab1bbe244964a08dac8fa18e5c2f4fa37/sentence.go#L77

Reply:
@icholy I have moved the splitting to `parseSentence` in 991edf1, however the unit tests still exclusively call `Parse`...

I think some test cases for handling sentences with tag blocks (i.e., the splitting code) should go to `sentence_test.go` and as you suggest `tagblocks_test.go` should focus on `parseTagBlock` directly.

Reply:
@icholy As per the discussion above, I've added tests for TAG Block handling in `parseSentence` (81eba81) and reworked the original TAG Block payload/content tests to target `parseTagBlock` directly instead of calling `Parse` with a full NMEA sentence (acffae7).